Fix tester test6

edouardproust opened the pull request:
- In `CgiHandler`, added env variable `SERVER_NAME` and `SERVER_PORT` dynamically.
- The rest of new variables is left as empty string (at least for now), because either too complex to implement (`REMOTE_ADDR` -> subject does not allow functions for getting client IP), or feature not implemented (Authentication: `AUTH_TYPE` and `REMOTE_USER`)
- Refactored Response  with `Header` type

edouardproust committed:
refactored Request and Network headers orga

edouardproust committed:
fix merge conflicts

edouardproust committed:
added env variables (some are just empty strings...)

edouardproust committed:
update

Skoteini-42 commented:
- Response has the headers in the opposite order now :

```
HTTP/1.1 200 OK
Connection: close
Content-Disposition: inline; filename="home.htm"
Content-Length: 8028
Content-Type: text/html
Date: Tue, 18 Nov 2025 17:53:21 GMT
Server: webserv/1.0
```

- I don't see anything affected by it so far, but here :

`_headers["connection"] = std::make_pair("Connexion", "keep-alive");`

Better write "Connection" with the same spelling, this is the British spelling what you used, but not written like this inside the protocol

I like how you made SERVER_NAME and SERVER_PORT dynamic! I only kept all these additional envs to make sure they won't be needed in a latter test, let's hope not, but again good to have and demonstrate, so very nice :)

edouardproust commented:
Yes they are in alphabetical order, true, so i rool back on this

edouardproust committed:
removed Request Header type logic

edouardproust committed:
merge with updated main

edouardproust committed:
updated /tests/website/home.htm

edouardproust commented:
- Rolled back on Request Header refactoring
- Fixed some issues in `isConnectionClose` methods in `Request` and `Response` (it seems ok, but run tests please to make sure)
- Updated `/tests/website/home.htm` following your changes to make it more clear with `PUT` and `HEAD` tests

edouardproust committed:
rolledback Header refactoring in Response

Skoteini-42 reviewed (approved):
Everything seems ok Ed, approving! Also very nice job with home.htm :)
